### ISSUE TITLE
Add unit testing

### ISSUE THREAD

**Issue by bomeara** [phylotastic/datelife#3]

**Comment by arlin** [comment]
@bomeara , according to what you said last week, unit tests have been added.  Is that right? If so, can we move this?

**Comment by bomeara** [comment]
Yes, they have. But I wanted to make sure we went through the proper review
process.

Best,
Brian

_______________________________________
Brian O'Meara
Associate Professor
Dept. of Ecology & Evolutionary Biology
U. of Tennessee, Knoxville
http://www.brianomeara.info

**Comment by bomeara** [comment]
Still waiting on review to close this out, I believe.

**Comment by arlin** [comment]
tell me where to look and I would be happy to take a look at it.

**Comment by bomeara** [comment]
Go to https://github.com/phylotastic/datelife and you'll see a build failing/passing badge. You can also look in the tests directory to see the unit tests. It is now the trend to look for code coverage, too; that's filed as a separate issue and isn't ready yet.

**Comment by arlin** [comment]
ProcessResultsList defaults to getting all taxa if is.null(taxa)?  Am I reading this right?

**Comment by bomeara** [comment]
It gets all taxa from final.matrices. Those were generated in an earlier
version of datelife but they're out of scope for that function. So that
behavior isn't good. Adding a ticket...

_______________________________________
Brian O'Meara
Associate Professor
Dept. of Ecology & Evolutionary Biology
U. of Tennessee, Knoxville
http://www.brianomeara.info

**Comment by arlin** [comment]
OK, got it.  My other comments are just about possible improvements and further tests.  
* line 37 of test_core.R takes the first tree off the list to make sure it parses out as class "phylo" when read by ape.   A further test would be to take the last tree from the list. 
* line 71 of datelife.R seems to take a roundabout approach to testing for the pattern "\\(.*\\).*;"
* there is quite a bit of complexity in ProcessInput that is not being tested in test_core.  I would suggest hard-coding a list of possible inputs with "A,B,C", "A, B, C", "(A,(B,C))" and so on (where A, B and C are taxon names with and without spaces or underscores in them), and then chewing through the list to make sure that ProcessInput gets each example right
* with regard to unmatchable taxa, I like Latin names better, e.g., Oscar the grouch is obviously genus Astronotus (Oscar) and the species would be 
lienosus (splenetic, grumpy).  But this might result in some false matches.  Fire-spewing dragon is Draco ignemvomentis .
